mypy/subtypes.py

```diff
@@ -549,7 +809,8 @@ def restrict_subtype_away(t: Type, s: Type) -> Type:
         # Since runtime type checks will ignore type arguments, erase the types.
         erased_s = erase_type(s)
         new_items = [item for item in t.relevant_items()
-                     if (not is_proper_subtype(erase_type(item), erased_s)
+                     if (not (is_proper_subtype(erase_type(item), erased_s) or
+                              is_proper_subtype(item, erased_s))
```

Comment:
This is not only related to protocols, but it affected several tests. This is actually a step in the direction of https://github.com/python/mypy/issues/3827. The logic is following, if I erase both ``item`` and ``s``, then there will be "too many" ``Any``s there, and ``is_proper_subtype`` will return ``False``, while at runtime ``isinstance()`` will actually return ``True`` for protocols. Maybe this is not 100% ideal, but it fixed several test failures without causing new failures in existing tests.

More generally, I think we could reconsider use of ``is_proper_subtype(erased_t, erased_s)`` for ``isinstance()`` with _nominal_ types, maybe there could be a better concept, something like ``is_runtime_subtype``? I don't think we need a separate issue for this, this is basically just an extended discussion for https://github.com/python/mypy/issues/3827